python/kserve/kserve/model.py

```diff
@@ -232,9 +237,10 @@ def postprocess(self, response: Union[Dict, InferResponse, ModelInferResponse],
         return response
 
     async def _http_predict(self, payload: Union[Dict, InferRequest], headers: Dict[str, str] = None) -> Dict:
-        predict_url = PREDICTOR_URL_FORMAT.format(self.predictor_host, self.name)
+        protocol = "https" if self.use_ssl else "http"
```

Comment:
Do we have test cases for this function to verify if http or https is used?

Reply:
Unfortunately not. I can't find a good way to add tests here. We have e2e tests for http but not for https.